doc/iterations/2024.3/shapes/overflow_label_strategy.adoc

```diff
@@ -0,0 +1,61 @@
+= Overflow label strategy
+
+== Problem
+
+In the existing node layouts, the inside label of a node will always be used to determine the minimum size.
+This means that in the case of a long label, the node will necessarily take a size that allows the label to be displayed without any modification.
+In some cases, specifier doesn't want that the label affects the node representation.
+
+== Key Result
+
+Allow the specifier to define a label that applies an auto wrap rather than enlarging the node.
+A label should be broken at any point if there are no otherwise-acceptable break points (like a long word or URL).
+
+== Solution
+
+Add to the view DSL, for inside label:
+
+* _maxLabelWidthExpression_
+* _overflowStrategy_: none, wrap or ellipsis
+
+If set, a label can only influence the size of its node up to _maxLabelWidthExpression_ result.
+There are two options for dealing with overflow situations, wrap text or display an ellipsis.
+With the _overflowStrategy_ _none_ option, the
+
+== Scenario
+
+- With a rectangular node style without child, a default width to 50 and a max label width to 100:
+* if a specifier declares an inside label with _wrap_ for _overflowStrategy_.
+** If the label is smaller than the node default width, then the node will be sized to match its default value:
++
+image:images/auto_wrap_label/node_with_small_label.png[label smaller than default node width]
++
+** If the label is larger than the node default width but smaller than the _maxLabelWidthExpression_ result, then the node will increase to display the whole label
++
+image:images/auto_wrap_label/node_with_normal_label.png[label smaller than the max label length]
++
+** If the label is larger than the _maxLabelWidthExpression_ result, then the node will be sized to match the _maxLabelWidthExpression_ result and the _overflowStrategy_ will be
+ applied on the label
++
+*** _none_ option
++
+image:images/auto_wrap_label/overflowStrategy_none_option.png[None option]
```

Comment:
This is considered as a bug by our users, if an inside label is not supposed to overflow from its parent node, then the size of the node has to be updated.

Reply:
I changed it to the current behavior, the node is expands to fit the label